golem-test-framework/src/components/component_service/mod.rs

```diff
@@ -90,6 +92,82 @@ pub trait ComponentServiceInternal: Send + Sync {
     fn plugin_wasm_files_service(&self) -> Arc<PluginWasmFilesService>;
+
+    async fn get_plugin_id(&self, name: &str, version: &str) -> crate::Result<Option<PluginId>> {
+        match self.plugin_client() {
+            PluginServiceClient::Grpc(mut client) => {
+                let response = client
+                    .get_plugin(GetPluginRequest {
+                        name: name.to_string(),
+                        version: version.to_string(),
+                    })
+                    .await?;
+                let converted = response.into_inner().result.and_then(|r| match r {
+                    get_plugin_response::Result::Success(result) => result
+                        .plugin
+                        .map(|p| PluginId::try_from(p.id.unwrap()).map_err(|e| anyhow!(e))),
+                    get_plugin_response::Result::Error(error) => Some(Err(anyhow!("{error:?}"))),
+                });
+                match converted {
+                    Some(Ok(inner)) => Ok(Some(inner)),
+                    Some(Err(e)) => Err(e),
+                    None => Ok(None),
+                }
+            }
+            PluginServiceClient::Http(_client) => {
+                // TODO: do a proper resolution here or split the clients
+                Ok(None)
+            }
+        }
+    }
+
+    async fn to_grpc_component(
+        &self,
+        component: golem_client::model::Component,
+    ) -> crate::Result<Component> {
+        let component = Component {
+            versioned_component_id: Some(VersionedComponentId {
+                component_id: Some(
+                    ComponentId(component.versioned_component_id.component_id).into(),
+                ),
+                version: component.versioned_component_id.version,
+            }),
+            component_name: component.component_name,
+            component_size: component.component_size,
+            metadata: Some(component.metadata.into()),
+            project_id: None,
+            account_id: None,
+            created_at: Some(SystemTime::from(component.created_at).into()),
+            component_type: Some(component.component_type as i32),
+            files: component
+                .files
+                .into_iter()
+                .map(|file| file.into())
+                .collect(),
+            installed_plugins: stream::iter(component.installed_plugins)
+                .then(async |install| {
+                    let (plugin_name, plugin_version) = match install.plugin {
+                        ReferencedPlugin::Registered(inner) => {
+                            (inner.plugin_name, inner.plugin_version)
+                        }
+                        _ => Err(anyhow!("reference to unregistered plugin"))?,
+                    };
+
+                    // TODO: This should error if this is none, but currently the http api cannot resolve the id, but the types force us to provide something here.
```

Comment:
This is an ugly side effect of us abstracting over the http and grpc api in the test framework (we have a number of these already).
For now the tests are happy, but we should give it some though how we want to improve the test framework when it comes to this.